include/solvers/simulation_parameters.h

```diff
@@ -133,6 +134,10 @@ class SimulationParameters
 
     Parameters::VelocitySource::declare_parameters(prm);
 
+    constrain_solid_domain.declare_parameters(
+      prm, 1); // at the moment, default value of the number of constraints is
+               // set to one since we are only applying to one-fluid simulations
```

Comment:
The default value should always be one right? We can use a similar mechanism to the one we use in the boundary conditions to pre-detect the number of contraints. Otherwise, you could have a max constraints to pregenerate them.